### Table of Contents:
* [1. Data Acquisition](#cell1)
* [2. Basic Insights of the Dataset](#cell2)


<a id="cell1"></a>
<h1 id="data_acquisition">1. Data Acquisition</h1>
<p>
There are various formats for a dataset like .csv, .json, .xlsx, etc. The dataset can be stored in different places, on your local machine or sometimes online. In this section, you will learn how to load a dataset into our Jupyter Notebook.<br>

In our case, the <a href="https://developer.ibm.com/exchanges/data/all/airline/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0151ENSkillsNetwork926-2022-01-01">Airline Dataset</a> is an online source, and it is in CSV (comma separated value) format. Let's use a smaller subset of the original dataset as an example to practice data reading.
<ul>
    <li>data source: <a href="https://dax-cdn.cdn.appdomain.cloud/dax-airline/1.0.1/lax_to_jfk.tar.gz" target="_blank">https://dax-cdn.cdn.appdomain.cloud/dax-airline/1.0.1/lax_to_jfk.tar.gz</a></li>
    <li>data type: csv</li>
</ul>

The Tidyverse library is a useful tool that enables us to read various datasets into a data frame; our Jupyter notebook platforms have a built-in <b>Tidyverse library</b> so that all we need to do is import Tidyverse without installing.
</p>

However, if you decide to run this on your local machine, you can use the below line of code to install Tidyverse before loading.


In [ ]:
# Uncomment to install tidyverse if running locally
#install.packages("tidyverse")

In [ ]:
# Load tidyverse
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ purrr::%||%()   masks base::%||%()
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


The Tidyverse library has a suite of packages that all work together well, namely:
* tidyr: helps create tidy data
* dplyr: manipulates and transforms data
* readr: easily read different types of datasets
* purrr: functional programming toolkit
* ggplot2: create plots and visualizations

In this notebook, we will mainly use functions from readr and dplyr. However, don't worry about memorizing which function is from which library, just know that all are included in Tidyverse.


<h2>Read Data</h2>
<p>
We use the function <code>readr::read_csv()</code> to read csv files. The <code>::</code> tells you which package the function is from, so <code>read_csv()</code> is from library <code>readr</code> (which is automatically loaded with Tidyverse). There are a few parts to the function to go over:
</p>

* `file` (or the first parameter): this is file path along with quotation marks, so that `read_csv()` will read the file into a data frame from that address. The file path can be either an URL or your local file address.
* `col_names`: by deafult this is set to `col_names = TRUE`. If this is TRUE then the first row is set as the headers (which is correct in this dataset).
* `col_types`: used to specify what types columns are. By default, `read_csv()` will guess the type of the columns but if there are columns you want to specify you can do so as well.
    * Posible types you could use are `col_logical()`, `col_integer()`, `col_number()`, `col_character()`. You can look at the documentation of `readr::cols` for more.
    
You can assign the loaded dataset to any variable name, here it is `sub_airline`.


In [ ]:
# url where the data is located
url <- "https://dax-cdn.cdn.appdomain.cloud/dax-airline/1.0.1/lax_to_jfk.tar.gz"
# download the file
download.file(url, destfile = "lax_to_jfk.tar.gz")
# untar the file so we can get the csv only
untar("lax_to_jfk.tar.gz", tar = "internal")
# read_csv only
sub_airline <- read_csv("lax_to_jfk/lax_to_jfk.csv",
                     col_types = cols(
                      'DivDistance' = col_number(),
                      'DivArrDelay' = col_number()
                      ))

Warning message in untar2(tarfile, files, list, exdir, restore_times):
“using pax extended headers”


After reading the dataset, there are a few functions you can use to get some initial information about the dataframe:
* `head(dataframe, n)`: returns the first *n* rows of the dataframe, if *n* is not specified, then by default the first 6 rows (not including the column headers) are returned
* `tail(dataframe, n)`: returns the last *n* rows of the dataframe, if *n* is not specified, then by default the last 6 rows are returned.
* `colnames(dataframe)`: retrieves the column names of the dataframe
* `dim(dataframe)`: retrieves the dimension (number of rows and columns) of the dataframe


In [ ]:
# show the first n = 3 rows
head(sub_airline, 3)

Month DayOfWeek FlightDate Reporting_Airline Origin Dest CRSDepTime
1  3    5         2003-03-28 UA                LAX    JFK  2210      
2 11    4         2018-11-29 AS                LAX    JFK  1045      
3  8    5         2015-08-28 UA                LAX    JFK  0805      
  CRSArrTime DepTime ArrTime ⋯ ArrDelayMinutes CarrierDelay WeatherDelay
1 0615       2209    0617    ⋯ 2               NA           NA          
2 1912       1049    1851    ⋯ 0               NA           NA          
3 1634       0757    1620    ⋯ 0               NA           NA          
  NASDelay SecurityDelay LateAircraftDelay DepDelay DepDelayMinutes DivDistance
1 NA       NA            NA                -1       0               NA         
2 NA       NA            NA                 4       4               NA         
3 NA       NA            NA                -8       0               NA         
  DivArrDelay
1 NA         
2 NA         
3 NA

Or, you can choose to not input `n`, then it will show the first or the last 6 rows as default.


In [ ]:
# show the first 6 rows
head(sub_airline)

Month DayOfWeek FlightDate Reporting_Airline Origin Dest CRSDepTime
1  3    5         2003-03-28 UA                LAX    JFK  2210      
2 11    4         2018-11-29 AS                LAX    JFK  1045      
3  8    5         2015-08-28 UA                LAX    JFK  0805      
4  4    7         2003-04-20 DL                LAX    JFK  2205      
5 11    3         2005-11-30 UA                LAX    JFK  0840      
6  4    1         1992-04-06 UA                LAX    JFK  1450      
  CRSArrTime DepTime ArrTime ⋯ ArrDelayMinutes CarrierDelay WeatherDelay
1 0615       2209    0617    ⋯ 2               NA           NA          
2 1912       1049    1851    ⋯ 0               NA           NA          
3 1634       0757    1620    ⋯ 0               NA           NA          
4 0619       2212    0616    ⋯ 0               NA           NA          
5 1653       0836    1640    ⋯ 0               NA           NA          
6 2308       1452    2248    ⋯ 0               NA           NA          
  NASDelay SecurityDelay LateAircraftDelay DepDelay DepDelayMinutes DivDistance
1 NA       NA            NA                -1       0               NA         
2 NA       NA            NA                 4       4               NA         
3 NA       NA            NA                -8       0               NA         
4 NA       NA            NA                 7       7               NA         
5 NA       NA            NA                -4       0               NA         
6 NA       NA            NA                 2       2               NA         
  DivArrDelay
1 NA         
2 NA         
3 NA         
4 NA         
5 NA         
6 NA

In [ ]:
# show the last 6 rows
tail(sub_airline)

Month DayOfWeek FlightDate Reporting_Airline Origin Dest CRSDepTime
1 10    6         1996-10-19 AA                LAX    JFK  0700      
2 10    3         2015-10-21 DL                LAX    JFK  2255      
3  7    5         2012-07-06 UA                LAX    JFK  0605      
4  1    3         2010-01-06 DL                LAX    JFK  2125      
5  5    5         2009-05-15 AA                LAX    JFK  2115      
6  7    4         2000-07-06 AA                LAX    JFK  2200      
  CRSArrTime DepTime ArrTime ⋯ ArrDelayMinutes CarrierDelay WeatherDelay
1 1509       0700    1610    ⋯ 61              NA           NA          
2 0720       2254    0723    ⋯  3              NA           NA          
3 1422       0600    1427    ⋯  5              NA           NA          
4 0543       2118    0540    ⋯  0              NA           NA          
5 0535       2119    0533    ⋯  0              NA           NA          
6 0622       2158    0610    ⋯  0              NA           NA          
  NASDelay SecurityDelay LateAircraftDelay DepDelay DepDelayMinutes DivDistance
1 NA       NA            NA                 0       0               NA         
2 NA       NA            NA                -1       0               NA         
3 NA       NA            NA                -5       0               NA         
4 NA       NA            NA                -7       0               NA         
5 NA       NA            NA                 4       4               NA         
6 NA       NA            NA                -2       0               NA         
  DivArrDelay
1 NA         
2 NA         
3 NA         
4 NA         
5 NA         
6 NA

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #1: </h1>
<b>Check the last 10 rows of data frame "sub_airline".</b>
</div>


In [ ]:
# Write your code below and press Shift+Enter to execute
tail(sub_airline, 10)

Month DayOfWeek FlightDate Reporting_Airline Origin Dest CRSDepTime
1   5    6         1988-05-14 AA                LAX    JFK  0830      
2   1    7         1994-01-02 UA                LAX    JFK  0730      
3   4    1         2000-04-17 UA                LAX    JFK  0900      
4   6    7         2003-06-08 DL                LAX    JFK  1230      
5  10    6         1996-10-19 AA                LAX    JFK  0700      
6  10    3         2015-10-21 DL                LAX    JFK  2255      
7   7    5         2012-07-06 UA                LAX    JFK  0605      
8   1    3         2010-01-06 DL                LAX    JFK  2125      
9   5    5         2009-05-15 AA                LAX    JFK  2115      
10  7    4         2000-07-06 AA                LAX    JFK  2200      
   CRSArrTime DepTime ArrTime ⋯ ArrDelayMinutes CarrierDelay WeatherDelay
1  1710       0829    1646    ⋯  0              NA           NA          
2  1548       0732    1549    ⋯  1              NA           NA          
3  1721       0859    1705    ⋯  0              NA           NA          
4  2052       1228    2052    ⋯  0              NA           NA          
5  1509       0700    1610    ⋯ 61              NA           NA          
6  0720       2254    0723    ⋯  3              NA           NA          
7  1422       0600    1427    ⋯  5              NA           NA          
8  0543       2118    0540    ⋯  0              NA           NA          
9  0535       2119    0533    ⋯  0              NA           NA          
10 0622       2158    0610    ⋯  0              NA           NA          
   NASDelay SecurityDelay LateAircraftDelay DepDelay DepDelayMinutes
1  NA       NA            NA                -1       0              
2  NA       NA            NA                 2       2              
3  NA       NA            NA                -1       0              
4  NA       NA            NA                -2       0              
5  NA       NA            NA                 0       0              
6  NA       NA            NA                -1       0              
7  NA       NA            NA                -5       0              
8  NA       NA            NA                -7       0              
9  NA       NA            NA                 4       4              
10 NA       NA            NA                -2       0              
   DivDistance DivArrDelay
1  NA          NA         
2  NA          NA         
3  NA          NA         
4  NA          NA         
5  NA          NA         
6  NA          NA         
7  NA          NA         
8  NA          NA         
9  NA          NA         
10 NA          NA

<details>
    <summary>Click here for the solution.</summary>

```r
tail(sub_airline, 10)
```
    
</details>


## Preprocess Data

Throughout this course, we are going to be focusing on flights from  LAX to JFK and predict the possibility of Amy's flight delays. So we also exclude flights that got cancelled or diverted. The previous example uses the final subset dataset `sub_airline`. However, there was some preprocessing done on the original dataset to get there. You can find the full dataset here: https://developer.ibm.com/technologies/artificial-intelligence/data/airline/.

From this original dataset, we will show you how we got the `sub_airline` dataset. After downloading the original data with `download.file()` and reading the file with `read_csv()`:

1. Firstly, we use `dplyr::filter()` to filter out all flights from `LAX` to `JFK`, and filter out the cancelled and diverted flights.
2. Then, we use `dplyr::select()` to select the columns by a pre-determined headers list. The original dataset contains more than 100 columns, so we are only going to use a few of the columns such as `Month`, `DaysOfWeek`, `FlightDate`, `Reporting_Airline`, and so on.

The below code is to show you how to subset a big dataset. The code is here for your reference. Feel free to play around with it and modify the columns. Remember **not** to run the below code in Skills Network Labs. The data size too big and it is better run to run locally so that Skills Network Labs does not freeze or lag.


Print and recheck our subset airline data using `dim()` on `sub_airline`.


This subset dataframe should include 2855 rows and 21 columns.


In [ ]:
# Check dimensions of the dataset
dim(sub_airline)

[1] 2855   21

 <div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #2: </h1>
<b>Find the name of the columns of the dataframe</b>
</div>


In [ ]:
# Write your code below and press Shift+Enter to execute
colnames(sub_airline)

[1] "Month"             "DayOfWeek"         "FlightDate"       
 [4] "Reporting_Airline" "Origin"            "Dest"             
 [7] "CRSDepTime"        "CRSArrTime"        "DepTime"          
[10] "ArrTime"           "ArrDelay"          "ArrDelayMinutes"  
[13] "CarrierDelay"      "WeatherDelay"      "NASDelay"         
[16] "SecurityDelay"     "LateAircraftDelay" "DepDelay"         
[19] "DepDelayMinutes"   "DivDistance"       "DivArrDelay"

<details>
    <summary>Click here for the solution.</summary>
    
```r
colnames(sub_airline)
```
    
</details>


<h2>Save Dataset</h2>
<p>
Correspondingly, "readr" enables us to save the dataset to csv by using the <code>write_csv()</code> method, you can add the file path and name along with quotation marks in the brackets.
</p>
<p>
For example, if you wanted to save the dataframe <b>sub_airline</b> as <b>lax_to_jfk.csv</b> to your local machine in the current directory, you can use the syntax below:
</p>


In [ ]:
write_csv(sub_airline, "lax_to_jfk.csv")

<h2>Read/Save Other Data Formats</h2>

We can also read and save other file formats, we can use similar functions to **`read_csv()`** and **`write_csv()`** for other data formats, the functions are listed in the following table:


| Data Format  | Read           | Save             |
| ------------- |:--------------:| ----------------:|
| csv           | `read_csv()`  |`write_csv()`     |
| tsv           | `read_tsv()`  |`write_tsv()`     |
| delimiter     |`read_delim()` |`write_delim()`   |
| ...           |   ...          |       ...        |

You can find more functions from the "readr" documentation [here](https://readr.tidyverse.org/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0151ENSkillsNetwork926-2022-01-01).


<a id="cell2"></a>
<h1 id="basic_insight">2. Basic Insights of the Dataset</h1>
<p>
After reading data into a dataframe, it is time for us to explore the dataset a little more.<br>
There are several ways to obtain essential insights of the data to help us better understand our dataset.
</p>


## Data Types

Data has a variety of types.

The main types stored in R are: `numeric`, `integer`, `complex`, `logical`, `character`. In order to better learn about each attribute, it is always good for us to know the data type of each column.

Let's check by using `sapply()` to apply  `typeof` to every column in the dataframe `sub_airline`:


In [ ]:
sapply(sub_airline, typeof)

Month         DayOfWeek        FlightDate Reporting_Airline 
         "double"          "double"          "double"       "character" 
           Origin              Dest        CRSDepTime        CRSArrTime 
      "character"       "character"       "character"       "character" 
          DepTime           ArrTime          ArrDelay   ArrDelayMinutes 
      "character"       "character"          "double"          "double" 
     CarrierDelay      WeatherDelay          NASDelay     SecurityDelay 
         "double"          "double"          "double"          "double" 
LateAircraftDelay          DepDelay   DepDelayMinutes       DivDistance 
         "double"          "double"          "double"          "double" 
      DivArrDelay 
         "double"

**Numeric and Integer Types**

Decimal values are called `numeric` in R. It is the default computational data type. If we assign a decimal value to a variable x as follows, x will be of numeric type.


In [ ]:
x = 10.5       # assign a decimal value
class(x)       # print the class name of x, which should be numeric

[1] "numeric"

Furthermore, even if we assign an integer to a variable k, it is still being saved as a numeric value.


In [ ]:
k = 1          # assign an integer value
class(x)       # print the class name of x, which should be numeric

[1] "numeric"

In order to create an integer variable in R, we invoke the integer function. We can be assured that y is indeed an integer by applying the is.integer function.


In [ ]:
y = as.integer(3)        # assign a integer value
class(y)                 # print the class name of y, which should be integer

[1] "integer"

**Complex Type**

A complex value in R is defined via the pure imaginary value $i$ (`0i` in R).


In [ ]:
z = 0i
class(z)

[1] "complex"

**Logical Type**

A logical value is often created via comparison between variables, such as True, False.


In [ ]:
logical_values = c(TRUE, T, FALSE, F)
class(logical_values)

[1] "logical"

**Character Type**

A character object is used to represent string values in R.


In [ ]:
class('this is a character')

[1] "character"

## Dplyr for Data Wrangling and Transformation

The dplyr package is very useful to transform data and get basic insights from the dataset. The most important functions are:
* `select()`
* `filter()`
* `summarize()`
* `arrange()`
* `mutate()`
* `group_by()`


### Pipe
Before we dive deeper into using some of the functions of dyplr, we first introduce the **pipe** operator **`%>%`**. The pipe operator allows us to chain together dplyr data wrangling functions.

An advantage of using pipe is that it replaces having messy nested functions that can be difficult to interpret like:


The same code can be written using pipe and makes it much easier to understand:


When interpreting the above lines of code, the pipe operator can be read as “then”.
The **`%>%`** operator allows us to go from one step in dplyr to the next easily, so the above example can clearily be interpreted as:
* **`filter`** our data frame for month 1 *then*
* **`group_by`** the reporting airline types *then*
* **`summarize`** the average `CarrierDelay` for each airline. The final output is the average `CarrierDelay` for each `Reporting_Airline` in month 1.


### Summarize

Now, let's go over some examples using basic dplyr functions with pipe.
If we would like to get a statistical summary of a column, such as count, column mean value, column standard deviation, etc., we can use the `summarize()` method (can also use `summarise()`).


This method will provide various summary statistics, excluding `NA` (Not Available) values.


In [ ]:
# group_by / summarize workflow example
sub_airline %>%
  group_by(Reporting_Airline) %>%
  summarize(avg_carrier_delay = mean(CarrierDelay, na.rm = TRUE)) # use mean value

Reporting_Airline avg_carrier_delay
1 AA                19.643478        
2 AS                14.833333        
3 B6                11.466667        
4 DL                24.912088        
5 HP                 3.333333        
6 PA (1)                  NaN        
7 TW                      NaN        
8 UA                14.509804        
9 VX                14.941176

The statistical metrics can tell the data scientist if there are mathematical issues that may exist in a particular column, such as extreme outliers and large deviations. The data scientist may have to address these issues later. The above example used `mean()` to return the mean value of arrival delay, you may also use `sd` for standard deviation, `median`, etc.


In [ ]:
# group_by / summarise workflow example
sub_airline %>%
  group_by(Reporting_Airline) %>%
  summarize(sd_carrier_delay = sd(CarrierDelay, na.rm = TRUE)) # use standard deviation

Reporting_Airline sd_carrier_delay
1 AA                36.047287       
2 AS                19.039831       
3 B6                29.040280       
4 DL                99.775375       
5 HP                 8.164966       
6 PA (1)                   NA       
7 TW                       NA       
8 UA                35.396821       
9 VX                39.813917

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #3: </h1>

Using `sub_airline`, get the mean of `ArrDelay` for each `Reporting_Airline`.
In other words, group by `Reporting_Airline` and summarize the mean of `ArrDelay` of each reporting airline. Remember to use `na.rm = TRUE`.
</div>


In [ ]:
sub_airline %>%
  group_by(Reporting_Airline) %>%
  summarize(airline_Delay = mean(ArrDelay, na.rm = TRUE))

Reporting_Airline airline_Delay
1 AA                 0.5939781   
2 AS                -1.1111111   
3 B6                 7.8139535   
4 DL                 5.0057034   
5 HP                13.2857143   
6 PA (1)            27.9696970   
7 TW                10.4270270   
8 UA                 3.8963093   
9 VX                 6.5193798

<details>
    <summary>Click here for the solution.</summary>

```r
sub_airline %>%
  group_by(Reporting_Airline) %>%
  summarize(airline_Delay = mean(ArrDelay, na.rm = TRUE))
```
    
</details>


### Glimpse
Another method you can use to check your dataset is:


It provide a concise summary of your DataFrame.


In [ ]:
# look at the info of airline dataset
glimpse(sub_airline)

Rows: 2,855
Columns: 21
$ Month             <dbl> 3, 11, 8, 4, 11, 4, 12, 12, 2, 3, 10, 8, 6, 2, 8, 3,…
$ DayOfWeek         <dbl> 5, 4, 5, 7, 3, 1, 1, 3, 4, 4, 4, 5, 4, 3, 7, 6, 4, 4…
$ FlightDate        <date> 2003-03-28, 2018-11-29, 2015-08-28, 2003-04-20, 200…
$ Reporting_Airline <chr> "UA", "AS", "UA", "DL", "UA", "UA", "VX", "HP", "UA"…
$ Origin            <chr> "LAX", "LAX", "LAX", "LAX", "LAX", "LAX", "LAX", "LA…
$ Dest              <chr> "JFK", "JFK", "JFK", "JFK", "JFK", "JFK", "JFK", "JF…
$ CRSDepTime        <chr> "2210", "1045", "0805", "2205", "0840", "1450", "104…
$ CRSArrTime        <chr> "0615", "1912", "1634", "0619", "1653", "2308", "190…
$ DepTime           <chr> "2209", "1049", "0757", "2212", "0836", "1452", "103…
$ ArrTime           <chr> "0617", "1851", "1620", "0616", "1640", "2248", "190…
$ ArrDelay          <dbl> 2, -21, -14, -3, -13, -20, 1, -40, 25, 7, 5, 34, 21,…
$ ArrDelayMinutes   <dbl> 2, 0, 0, 0, 0, 0, 1, 0, 25, 7, 5, 34, 21, 0, 14, 682…
$ CarrierDelay  

<p>
Here we are able to see the information of our dataframe, it prints out column names, column types, and first few values for brief data previews. It also shows us this subset dataset has 2,855 rows and 21 columns.
</p>

This subset dataset `sub_airline` ("lax_to_jfk.csv") will be used throughout the course.


<h1>Excellent! You have just completed the  Introduction  Notebook!</h1>


### About the Authors:

This notebook was written by [Yiwen Li](https://developer.ibm.com/profiles/yiwen.li/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0151ENSkillsNetwork926-2022-01-01) and [Gabriela De Queiroz](https://developer.ibm.com/profiles/gdq/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0151ENSkillsNetwork926-2022-01-01).

<p><a href="https://www.linkedin.com/in/yiwen-li-47a019119/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0151ENSkillsNetwork926-2022-01-01" target="_blank">Yiwen Li</a> has approximately three year experiences in big tech industry. Currently, she is a developer advocate, a data scientist, a product manager at IBM, where she designs and develops data science solutions and Machine Learning models to solve real world problems. She has delivered talks this year in JupyterCon, PyCon, Pyjamas, CrowdCast.ai, Global AI on Tour 2020 and Belpy 2021 with hundreds of attendants per talk.
        
<a href="https://www.linkedin.com/in/gabrieladequeiroz/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0151ENSkillsNetwork926-2022-01-01" target="_blank">Gabriela de Queiroz</a> is a Sr. Engineering & Data Science Manager at IBM where she manages and leads a team of developers working on Data & AI Open Source projects. She works to democratize AI by building tools and launching new open source projects.
She is the founder of AI Inclusive, a global organization that is helping increase the representation and participation of gender minorities in Artificial Intelligence. She is also the founder of R-Ladies, a worldwide organization for promoting diversity in the R community with more than 190 chapters in 50+ countries.
She has worked in several startups and where she built teams, developed statistical models, and employed a variety of techniques to derive insights and drive data-centric decisions


Copyright © 2021 IBM Corporation. All rights reserved.
